src/libaktualizr/crypto/p11engine.cc

```diff
@@ -127,20 +128,34 @@ P11Engine::P11Engine(P11Config config) : config_(std::move(config)), ctx_(config
 }
 
 boost::filesystem::path P11Engine::findPkcsLibrary() {
-#ifdef TEST_PKCS11_ENGINE_PATH
-  const boost::filesystem::path custom_path = TEST_PKCS11_ENGINE_PATH;
+#ifdef PKCS11_ENGINE_PATH
+  const boost::filesystem::path custom_path = PKCS11_ENGINE_PATH;
 #else
   const boost::filesystem::path custom_path;
 #endif
-  const boost::filesystem::path openssl11_path = "/usr/lib/engines-1.1/pkcs11.so";
-  const boost::filesystem::path default_path = "/usr/lib/engines/pkcs11.so";
-  if (boost::filesystem::exists(custom_path)) {
-    return custom_path;
-  } else if (boost::filesystem::exists(openssl11_path)) {
-    return openssl11_path;
+  static const std::array<boost::filesystem::path, 3> engine_system_paths = {
+    "/usr/lib/engines-1.1/pkcs11.so",
+    "/usr/lib/engines/pkcs11.so",
+    "/usr/lib/x86_64-linux-gnu/engines-1.1/pkcs11.so"
```

Comment:
x86_64-linux-gnu? This is not generic across architectures.

Reply:
Yes, this is a small sample of possible locations I've seen (in this case, Ubuntu Bionic x86-64).

The recommended way should be to define `PKCS11_ENGINE_PATH`, this is only a helpful fallback. But maybe this is more harmful than useful?

Reply:
Problem is that if you want to cover other common locations this list will end up growing quite a bit as it depends on the distro and architecture used (e.g. bionic armhf/arm64/ppc64 would all have a different linux-gnu path).

Shouldn't necessarily be a problem, but having PKCS11_ENGINE_PATH is probably cleaner.

Reply:
Yes, I agree, that's why the list is not intended to be exhaustive, just be helpful for the common desktop testing scenario.

I will at least add some comments or maybe even do that through CMake in a safer way.